File: UsingProcessLifecycleOwnerForLibs/lib/src/main/java/github/nisrulz/sample/lib/MyLib.java
package github.nisrulz.sample.lib;

import android.arch.lifecycle.DefaultLifecycleObserver;
import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.LifecycleOwner;
import android.support.annotation.NonNull;

public class MyLib implements DefaultLifecycleObserver {

    public MyLib(final Lifecycle lifecycle) {
        lifecycle.addObserver(this);
    }

    @Override
    public void onCreate(@NonNull final LifecycleOwner owner) {
        System.out.println("LifecycleApplication.onCreate");
    }

    @Override
    public void onStart(@NonNull final LifecycleOwner owner) {
        System.out.println("LifecycleApplication.onStart");

    }

    @Override
    public void onResume(@NonNull final LifecycleOwner owner) {
        System.out.println("LifecycleApplication.onResume");
    }

    @Override
    public void onPause(@NonNull final LifecycleOwner owner) {
        System.out.println("LifecycleApplication.onPause");
    }

    @Override
    public void onStop(@NonNull final LifecycleOwner owner) {
        System.out.println("LifecycleApplication.onStop");
    }

    @Override
    public void onDestroy(@NonNull final LifecycleOwner owner) {
        System.out.println("LifecycleApplication.onDestroy");
    }
}

File: UsingProcessLifecycleOwnerForLibs/lib/src/main/java/github/nisrulz/android_lib/MyLib.kt
package github.nisrulz.android_lib

import androidx.lifecycle.DefaultLifecycleObserver
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleOwner

class MyLib(lifecycle: Lifecycle) : DefaultLifecycleObserver {
    override fun onCreate(owner: LifecycleOwner) {
        println("LifecycleApplication.onCreate")
    }

    override fun onStart(owner: LifecycleOwner) {
        println("LifecycleApplication.onStart")
    }

    override fun onResume(owner: LifecycleOwner) {
        println("LifecycleApplication.onResume")
    }

    override fun onPause(owner: LifecycleOwner) {
        println("LifecycleApplication.onPause")
    }

    override fun onStop(owner: LifecycleOwner) {
        println("LifecycleApplication.onStop")
    }

    override fun onDestroy(owner: LifecycleOwner) {
        println("LifecycleApplication.onDestroy")
    }

    init {
        lifecycle.addObserver(this)
    }
}